io/pci/PowerNVEEH.py

```diff
@@ -0,0 +1,218 @@
+                                                      num_of_hit))
+                                break
+                            else:
+                                self.log.info("PE recovered successfully")
+                else:
+                    self.log.warning("EEH inject failed for 5 times with\
+                               function %s" % func)
+                    enter_loop = False
+                    break
+            if not enter_loop:
+                break
+        else:
+            if self.check_eeh_removed():
+                self.log.info("PE %s removed successfully" % self.pe)
+            else:
+                self.fail("PE %s not removed after max hit" % self.pe)
+
+    def basic_eeh(self, addr, err, func, phb, pe):
+
+        """
+        Injects Error, and checks for PE recovery
+        returns True, if recovery is success, else Flase
+        """
+        self.clear_dmesg_logs()
+        return_code = self.error_inject(addr, err, func, phb, pe)
+        if return_code != EEH_HIT:
+            self.log.info("Skipping verification, as command failed")
+        if not self.check_eeh_hit():
+            self.log.info("PE %s EEH hit failed" % self.pe)
+            return EEH_MISS
+        else:
+            self.log.info("PE %s EEH hit success" % self.pe)
+            return EEH_HIT
+
+    @classmethod
+    def error_inject(cls, addr, err, func, phb, pe):
+
+        """
+        Form a command to inject the error
+        """
+        cmd = "echo %s:%s:%s:0:0 > /sys/kernel/debug/powerpc/PCI%s/err_injct \
+               && lspci -ns %s" % (addr, err, func, phb, pe)
+        res = process.system_output(cmd, ignore_status=True, shell=True)
+        return int(res[-1])
+
+    def check_eeh_pe_recovery(self):
+
+        """
+        Check if the PE is recovered successfully after injecting EEH
+        """
+        cmd = "dmesg | grep -i 'EEH: Notify device driver to resume'; echo $?"
+        tries = 60
+        for _ in range(0, tries):
+            res = process.system_output(cmd, ignore_status=True, shell=True)
+            if int(res[-1]) != 0:
+                self.log.info("waiting for PE to recover %s" % self.pe)
+                time.sleep(1)
+            else:
+                break
+        else:
+            raise EEHRecoveryFailed("EEH recovery failed", self.pe)
+        tries = 30
+        for _ in range(0, tries):
+            for device in pci.get_pci_addresses():
+                if self.pe in device:
+                    return True
+                time.sleep(1)
+            return False
+
+    @classmethod
+    def clear_dmesg_logs(cls):
+
+        """
+        Clears dmesg logs, so that functions which uses dmesg
+        gets the latest logs
+        """
+        cmd = "dmesg -C"
+        process.system(cmd, ignore_status=True, shell=True)
+
+    @classmethod
+    def check_eeh_hit(cls):
+
+        """
+        Function to check if EEH is successfully hit
+        """
+        tries = 10
```

Comment:
Isn't 10 sec too less time here ? Can't it be 30 sec at least ?
And also in PowerVMEEH.